ext/mbstring/libmbfl/filters/mbfilter_utf7.c

```diff
@@ -562,8 +564,10 @@ static size_t mb_utf7_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf
 			}
 			out = handle_utf16_cp((n3 << 14) | (n4 << 8) | (n5 << 2) | ((n6 & 0x30) >> 4), out, &surrogate1);
 			if (p == e) {
-				if ((n6 & 0xF) || surrogate1)
+				if ((n6 & 0xF) || surrogate1) {
 					*out++ = MBFL_BAD_INPUT;
+					surrogate1 = 0;
```

Comment:
I caught below on lldb. Could you add test case?

```
mb_convert_encoding("+999euJ", "UTF-8", "UTF-7");
```

Reply:
Sure, thanks.